src/blocks/EmbedBlock/EmbedBlock.astro

```diff
@@ -0,0 +1,58 @@
+---
+import type { EmbedBlockFragment } from '@lib/types/datocms';
+import { fetchPageTitle, getOEmbedProvider } from './index';
+import type { OEmbedAny } from './index';
+import DefaultEmbed from './embeds/Default.astro';
+import TwitterEmbed from './embeds/Twitter.astro';
+import VimeoEmbed from './embeds/Vimeo.astro';
+import YouTubeEmbed from './embeds/YouTube.astro';
+
+interface Props {
+  block: EmbedBlockFragment;
+}
+const { block } = Astro.props;
+const { data, url }: { data: OEmbedAny; url: string } = block;
+const fallbackText = data ? null : await fetchPageTitle(block.url);
+const providerName = (data || getOEmbedProvider(url))?.provider_name;
+const isProvider = (name: string) =>
+  providerName?.toLowerCase() === name.toLowerCase();
+---
+
+<div class='embed-block'>
+  {
+    data && data.html ? (
+      isProvider('Twitter') ? (
+        <TwitterEmbed {data} {url} />
+      ) : isProvider('Vimeo') ? (
+        <VimeoEmbed {data} {url} />
+      ) : isProvider('YouTube') ? (
+        <YouTubeEmbed {data} {url} />
+      ) : (
+        <DefaultEmbed {data} {url} />
+      )
+    ) : (
+      <a
+        class='embed-fallback'
+        href={block.url}
+        target='_blank'
+        rel='noopener noreferrer'
+      >
+        [ unable to load from {providerName} ]<br />
```

Comment:
Is this fallback still needed? Maybe a basic version when no `data.html` is available? If we keep this fallback, then this line should be translated.